app/main.py

```
@@ -1,52 +1,38 @@
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+_SERIALIZERS = {
+    "json": JSONSerializer(),
+    "xml": XMLSerializer(),
+}
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
-        if cmd == "display":
-            book.display(method_type)
-        elif cmd == "print":
-            book.print_book(method_type)
-        elif cmd == "serialize":
-            return book.serialize(method_type)
+        try:
+            if cmd == "display":
+                _DISPLAYS[method_type].display(book)
+            elif cmd == "print":
+                _PRINTERS[method_type].print_book(book)
+            elif cmd == "serialize":
+                return _SERIALIZERS[method_type].serialize(book)
+        except KeyError as error:
+            raise ValueError(
+                f"Unknown command or method type: {cmd}/{method_type}"
+            ) from error
+    return None
```

The command loop silently ignores unknown `cmd` values because there's no `else` branch; only KeyError from dictionary lookup is turned into a ValueError. If the intent is to treat unrecognized commands as errors, add an explicit `else: raise ValueError(...)` so behaviour is explicit and consistent. Also note that `serialize` returns immediately (early return) which preserves existing behaviour but should be documented in a comment if this early exit is important to callers/tests.